packages/back-end/src/services/experimentNotifications.ts

```diff
@@ -0,0 +1,208 @@
+    : (experiment.pastNotifications || []).filter((t) => t !== type);
+
+  await updateExperiment({
+    experiment,
+    context,
+    changes: {
+      pastNotifications,
+    },
+  });
+};
+
+export const notifyAutoUpdate = ({
+  context,
+  experiment,
+  success,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  success: boolean;
+}) =>
+  memoizeNotification({
+    context,
+    experiment,
+    type: "auto-update",
+    triggered: !success,
+    dispatch: () =>
+      dispatchEvent(context, {
+        type: "auto-update",
+        success,
+        experimentId: experiment.id,
+        experimentName: experiment.name,
+      }),
+  });
+
+export const MINIMUM_MULTIPLE_EXPOSURES_PERCENT = 0.01;
+
+const notifyMultipleExposures = async ({
+  context,
+  experiment,
+  firstResult,
+  snapshot,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  firstResult: ExperimentReportResultDimension;
+  snapshot: ExperimentSnapshotDocument;
+}) => {
+  const totalsUsers = firstResult.variations.reduce(
+    (totalUsersCount, { users }) => totalUsersCount + users,
+    0
+  );
+  const percent = snapshot.multipleExposures / totalsUsers;
+  const multipleExposureMinPercent =
+    context.org.settings?.multipleExposureMinPercent ??
+    MINIMUM_MULTIPLE_EXPOSURES_PERCENT;
+
+  const triggered = multipleExposureMinPercent <= snapshot.multipleExposures;
+
+  await memoizeNotification({
+    context,
+    experiment,
+    type: "multiple-exposures",
+    triggered,
+    dispatch: async () => {
+      if (!triggered) return;
+
+      await dispatchEvent(context, {
+        type: "multiple-exposures",
+        experimentId: experiment.id,
+        experimentName: experiment.name,
+        usersCount: snapshot.multipleExposures,
+        percent,
+      });
+    },
+  });
+};
+
+export const DEFAULT_SRM_THRESHOLD = 0.001;
+
+const notifySrm = async ({
+  context,
+  experiment,
+  firstResult,
+}: {
+  context: Context;
+  experiment: ExperimentInterface;
+  firstResult: ExperimentReportResultDimension;
+}) => {
+  const srmThreshold =
+    context.org.settings?.srmThreshold ?? DEFAULT_SRM_THRESHOLD;
+
+  const triggered = srmThreshold <= firstResult.srm;
+
+  await memoizeNotification({
+    context,
+    experiment,
+    type: "srm",
+    triggered,
+    dispatch: async () => {
+      if (!triggered) return;
+
+      await dispatchEvent(context, {
+        type: "srm",
+        experimentId: experiment.id,
+        experimentName: experiment.name,
+        threshold: srmThreshold,
+      });
+    },
+  });
+};
+
+export const notifyExperimentChange = async ({
+  context,
+  snapshot,
+}: {
+  context: Context;
+  snapshot: ExperimentSnapshotDocument;
+}) => {
+  const experiment = await getExperimentById(context, snapshot.experiment);
+  if (!experiment) throw new Error("Error while fetching experiment!");
+
+  const [
+    {
+      results: [firstResult],
+    },
+  ] = snapshot.analyses.sort(
```

Comment:
I think this should work, but I meant literally that we should get the `0`th indexed analysis. e.g. `snapshot.analyses[0]`

Reply:
Done and factored out. Thanks for the help!